docs/release.md

```diff
@@ -0,0 +1,34 @@
+# Release
+This document describes how to release the ClusterAPI BYOH infrastructure provider.
+
+
+### Current BYOH release: `v0.3.0`
+
+---
+##### Follow below steps to make a new release in BYOH
+1. Update `metadata.yaml` file if a new major/minor version is being released. This is not needed in case of patch release. Map the newly added version to supported ClusterAPI contract version.
```

Comment:
Suggestion: Update the filename to `cluster-api-provider-bringyourownhost/metadata.yaml`
```suggestion
1. Open a PR to update `cluster-api-provider-bringyourownhost/metadata.yaml` file if a new major/minor version is being released. This is not needed in case of patch release. Map the newly added version to supported ClusterAPI contract version.
```